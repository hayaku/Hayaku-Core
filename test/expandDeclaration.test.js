// Proxy for jest test to write abbr only once, ok
const test = (desc, func) => it(desc, () => func(desc));
test.only = (desc, func) => it.only(desc, () => func(desc));
test.skip = (desc, func) => it.skip(desc, () => func(desc));

const Hayaku = require('./../');
const getDeclaration = abbr => {
  const results = Hayaku.expandDeclaration(abbr);
  return results[0] && `${results[0].property}: ${results[0].value}`;
}

describe('simple declarations test:', () => {
  test('pos:', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('pos', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('poa', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('por', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('fl', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('fr', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('f', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('f:l', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('dn', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('bgn', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('curp', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('wa', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('bsbb', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
});

describe('some suddenly complex declarations test:', () => {
  test('fs', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('fz', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('fv', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('fv', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('ti', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('lh', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('orp', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('rz', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('cur', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('op', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('fst', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
});

describe('declarations with numeric values:', () => {
  test('w10', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('w1.5', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('w25%', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('w50p', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('w20px', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
  test('w20e', abbr => expect(getDeclaration(abbr)).toMatchSnapshot());
});
