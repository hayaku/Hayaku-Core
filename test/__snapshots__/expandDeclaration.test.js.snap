// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`declarations with numeric values: w1.5 1`] = `"width: 1.5em"`;

exports[`declarations with numeric values: w10 1`] = `"width: 10px"`;

exports[`declarations with numeric values: w20e 1`] = `"width: 20em"`;

exports[`declarations with numeric values: w20px 1`] = `"width: 20px"`;

exports[`declarations with numeric values: w25% 1`] = `"width: 25%"`;

exports[`declarations with numeric values: w50p 1`] = `"width: 50%"`;

exports[`simple declarations test: bgn 1`] = `"background: none"`;

exports[`simple declarations test: bsbb 1`] = `"box-sizing: border-box"`;

exports[`simple declarations test: curp 1`] = `"cursor: pointer"`;

exports[`simple declarations test: dn 1`] = `"display: none"`;

exports[`simple declarations test: f 1`] = `"float: undefined"`;

exports[`simple declarations test: f:l 1`] = `"float: left"`;

exports[`simple declarations test: fl 1`] = `"float: left"`;

exports[`simple declarations test: fr 1`] = `"float: right"`;

exports[`simple declarations test: poa 1`] = `"position: absolute"`;

exports[`simple declarations test: por 1`] = `"position: relative"`;

exports[`simple declarations test: pos 1`] = `"position: static"`;

exports[`simple declarations test: pos: 1`] = `"position: undefined"`;

exports[`simple declarations test: wa 1`] = `"width: auto"`;

exports[`some suddenly complex declarations test: cur 1`] = `"cursor: undefined"`;

exports[`some suddenly complex declarations test: fs 1`] = `"font-size: undefined"`;

exports[`some suddenly complex declarations test: fst 1`] = `"font-style: undefined"`;

exports[`some suddenly complex declarations test: fv 1`] = `"font-variant: undefined"`;

exports[`some suddenly complex declarations test: fv 2`] = `"font-variant: undefined"`;

exports[`some suddenly complex declarations test: fz 1`] = `"font-size: undefined"`;

exports[`some suddenly complex declarations test: lh 1`] = `"line-height: undefined"`;

exports[`some suddenly complex declarations test: op 1`] = `"opacity: undefined"`;

exports[`some suddenly complex declarations test: orp 1`] = `"orphans: undefined"`;

exports[`some suddenly complex declarations test: rz 1`] = `"resize: undefined"`;

exports[`some suddenly complex declarations test: ti 1`] = `"text-indent: undefined"`;
